### ISSUE TITLE
Why do we need to use learning rate scheduler while using BERT?

### ISSUE THREAD

**Issue by cemilcengiz** [allenai/allennlp#3677]
I am trying to train a BERT model on SRL, but could not understand why we would need a scheduler while using BERT since we already have an optimizer with an adaptive learning rate as well as warm up and linear decay. For example, although [https://github.com/allenai/allennlp/blob/master/training_config/bert_base_srl.jsonnet](url) has already bert_adam (or huggingface_adamw in master branch) optimizer, it also contains a slanted_triangular scheduler. Would you please explain why we need this?

Also, in the same bert_base_srl config file, "num_steps_per_epoch" is set to 8829. Can you also explain how we can get this number? In my case, when I load Ontonotes 5.0 SRL dataset, I get around 278030 instances. Since the example config has batch size of 32, I divide that by 32 and get 8688, which is not very close to the one in the config file. Finally, if I have 3 gpus should I divide this number by 3 while setting it in the config (P.S: I am using v0.9)? Thank you.

**Comment by DeNeutoy** [comment]
1. `AdamW` in huggingface no longer includes a learning rate schedule: https://github.com/huggingface/transformers/blob/master/src/transformers/optimization.py#L96

2. In previous versions, we disabled the schedule that used to be in AdamW (I cannot remember why we did this, I think there was some bug in it originally or something)

3. I don't know why you get a different number of instances, but yes it is the number of gradient updates per epoch, ie instances/batch_size.

4. No, allennlp v0.9 splits instances across the batch dimension when using multiple gpus, so you do not need to further divide it.

**Comment by brendan-ai2** [comment]
@DeNeutoy, I may be misunderstanding your point 4), but I don't believe that's correct. In 0.9 we simply take a batch for each GPU. See https://github.com/allenai/allennlp/blob/052353ed62e3a54fd7b39a660e65fc5dd2f91c7d/allennlp/training/trainer.py#L297. So I think dividing by the number of GPUs is appropriate.

**Comment by DeNeutoy** [comment]
Yes you are right, sorry.

**Comment by brendan-ai2** [comment]
I think this has been addressed. Please comment again if anything comes up.
